### PR TITLE
window.gpu -> navigator.gpu

### PR DESCRIPTION
**Pull request by kainino0x**
This is more of an idea than a proposal. When looking at some of the other APIs (WebUSB in particular) it seemed like this was a more natural API. @beaufortfrancois, since you proposed the thing we have currently, what do you think of this?

### CODE REVIEW

**Comment by beaufortfrancois** [comment]
WebUSB uses `navigator.usb.requestAdapter()`, not `navigator.requestUSBAdapter()`.

I'd be if favor of moving it from `window` to `navigator` to follow other recent Web APIs such as [WebXR](https://github.com/immersive-web/webxr/blob/master/explainer.md),  [WebBluetooth](https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetooth-requestdevice), [WebUSB](https://wicg.github.io/webusb/#dom-usb-requestdevice), and WIP [Serial API](http://wicg.github.io/serial/).

```js
await navigator.xr.requestSession(...);
await navigator.bluetooth.requestDevice(...);
await navigator.bluetooth.requestLEScan(...);
await navigator.usb.requestDevice(...);
await navigator.serial.requestPort(...);
```

Older Web APIs such as [EME](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/requestMediaKeySystemAccess) and [WebMIDI](https://webaudio.github.io/web-midi-api/#dom-navigator-requestmidiaccess) although used the other pattern you've mentioned:

```js
await navigator.requestMediaKeySystemAccess(...);
await navigator.requestMIDIAccess();
```

**Comment by kainino0x** [comment]
I got the `navigator.requestGPUAdapter` idea from a few of the other apis (getGamepads, getBattery, and the two you mentioned) - just had forgotten which they were.

The thing about having `navigator.gpu.requestAdapter` is that with WebUSB, navigator.usb is an instance of an interface - an event target which exposes onconnect/ondisconnect. With WebGPU, there's no need to add a global interface GPU and have an instance of it on `navigator.gpu`, so I used the other pattern.

However, `navigator.bluetooth` does seem to be an empty interface. So it would probably make sense to copy that pattern.

**Comment by kainino0x** [comment]
(Actually, it turns out that `navigator.bluetooth` is not empty. When I checked in Dev Tools I forgot to look at its __proto__.)

**Comment by kainino0x** [comment]
After looking at [`navigator.xr`](https://immersive-web.github.io/webxr/#xr-interface) they don't have much there, but they _do_ have `ondevicechange`. In the future there's a decent chance that WebGPU will want `onadapterchange` too, so it probably makes sense to have a place to put it. Uploading a new revision.

File: design/sketch.webidl [hunk 1]
```diff
@@ -815,10 +815,20 @@ dictionary GPURequestAdapterOptions {
 };
 
 [Exposed=Window]
```
Comment (Kangz) [inline design/sketch.webidl]: Shouldn't moving `gpu` to `navigator` just require changing `Exposed=Window` to `Exposed=Navigator`?

Comment (kainino0x) [reply design/sketch.webidl]: Hm, I copied this from WebUSB and Web Bluetooth, but I don't see anything in Web IDL that would prevent it.

**Comment by kainino0x** [comment]
Agreed on the call to move `gpu` to the navigator; just need to work out the exact mechanics of that change on this PR.

**Comment by grorg** [comment]
Discussed at [25 March teleconference](https://docs.google.com/document/d/1vDcZYv8dfJI-Rilbqc_4OJKjWhzDUPNdoa0dp_TGssc/edit)

**Comment by jdashg** [comment]
Merge after rebase!

**Pull request merged**